### ISSUE TITLE
subscribed returns true after subscription trail expiration

### ISSUE THREAD

**Issue by ramakrishnamundru** [laravel/cashier#486]
The `subscribed` method checks if the subscription is valid
https://github.com/laravel/cashier/blob/7.0/src/Billable.php#L187,
but the `valid` method returns `true` even after the trail expiration, because the `active` method returns `true` because we don't set `ends_at` on the subscription for trails.
https://github.com/laravel/cashier/blob/7.0/src/Subscription.php#L70-L83
one possible solution I can think of is to set `ends_at` along with `trial_ends_at`, or add another check for expired trails.

**Comment by driesvints** [comment]
So the way this works is that the `ends_at` is meant to denounce at which point the subscription is cancelled. As long as it's `null` it's a subscription that's active and ongoing. The `subscribed` method just checks if a user has an active subscription which still will return true even after the trial has expired. If you want to check wether the user is past their trial period you should use `$user->subscription('main')->onTrial()`.

So basically this is intended behavior.

https://laravel.com/docs/5.7/billing#subscription-trials
